src/Provider/GeocodeEarth/GeocodeEarth.php

```diff
@@ -80,7 +87,13 @@ public function reverseQuery(ReverseQuery $query): Collection
         $coordinates = $query->getCoordinates();
         $longitude = $coordinates->getLongitude();
         $latitude = $coordinates->getLatitude();
-        $url = sprintf(self::REVERSE_ENDPOINT_URL, $latitude, $longitude, $this->apiKey, $query->getLimit());
+
+        $url = $this->root.'/v'.$this->version.'/reverse'.'?'.http_build_query([
```

Comment:
Would it be worth storing the whole URL (minus the query params) rather than concatenating it here and in the `geocodeQuery`. Maybe could be done in the constructor?